### ISSUE TITLE
Create a set of UI mockups

### ISSUE THREAD

**Issue by ragingsquirrel3** [alliance-genome/agr_prototype#1]
I plan to create some UI mockups of the search results page to further explore the desired use cases in a visual format.  Show as table and list.

**Comment by ragingsquirrel3** [comment]
<img width="1373" alt="agr_mock_step_1" src="https://cloud.githubusercontent.com/assets/1258388/19093979/753f0dde-8a42-11e6-9c1a-033d0c3dad75.png">
<img width="1371" alt="agr_mock_step2" src="https://cloud.githubusercontent.com/assets/1258388/19093978/753eae98-8a42-11e6-8bf9-f247716a9757.png">
<img width="1374" alt="agr_mock_step3" src="https://cloud.githubusercontent.com/assets/1258388/19093980/75516baa-8a42-11e6-9d82-6b2809143536.png">

All the data here is fake.

Here are some mockups which show how a user would search within the "gene" category using the query "huntington's."  In this case it gives 5 result genes which have a hit in the disease field (how that field gets created is still in flux).  The assumption is that genes have a list of 0 or more diseases connected to them and that field gets searched.  

In step 2, the user clicks the species filter to select mouse genes, and in step 3 they click on the table view to see the results as a table.  

Just to clarify, the facets (or filters) on the left are not going to GO pages or disease pages.  Rather, they filter the initial set of genes by the indicated values.  Clicking "cell periphery" would filter results to genes annotated to  "cell periphery."  A point of discussion may be whether GO could be split into cellular component, biological process and molecular function.

**Comment by monicacecilia** [comment]
Thanks for posting this. Still exploring, but I have one comment that comes to mind. This is WRT displaying GO terms: 
I think the filters for GO can be kept as they are currently displayed. I suspect that at this point users will be interested in exploring all available classes for GO terms at a level of detail that is more meaningful than the three ontologies.
Alternatively (and this is perhaps what you already have in mind) I also imagine you could fold the children terms and the number of 'items' (e.g. genes) associated with them within each of those three classes and display them that way.

**Comment by cmungall** [comment]
Yes, to follow up on @monicacecilia's point, I agree the counts should be not only for the direct terms, but also for inferred annotations too. For example, 'cell periphery' would include genes annotated to 'chorion'. The closure rules are not always straightforward, we use an OWL reasoner as part of the process of building the golr closure tables (this will be increasingly important as we move to more expressive GO annotations), cc @kltm

**Comment by doughowe** [comment]
One thing we noticed when creating facet values with counts from the closure for each term is that facet value lists ordered by associated data volume result in only high level ontology terms being shown...often too high to be useful...like 'whole organism' for anatomy or "molecular function" for GO.  I think we tried to mitigate this by having some high level terms excluded from the facet value lists.  That worked but has had a few minor undesirable effects elsewhere in our system where we DID want to include terms like "whole organism".  Food for thought...

**Comment by kltm** [comment]
@doughowe For different use cases (and therefor pages), we have different closures that we expose to the user. Which closure is exposed in which interface is taken care of by a YAML file that drives the widgets.
This is all part of a larger indexing pattern that we make use of to try and as much code as possible reusable in AmiGO and other external applications.

**Comment by cmungall** [comment]
As a concrete example try the species facet in annotation search - the user sees a small configurable subset of above-species-level taxon groupings in the facet, but the API user can use any node from NCBITaxon

**Comment by ragingsquirrel3** [comment]
We can still discuss this here but I'm going to close the issue to note that "mockups are posted."

**Comment by doughowe** [comment]
Cool, thanks!  It looks like a similar filter isn't applied for the GO Class facet?  The top maybe 7 facet values are the kind we've been trying to avoid showing because they are of low value (IMO) in this context.  The problem was a little more significant in the ZFIN faceted search because we are only showing about 5 values by default...so those dubious high level terms are all you would get!

**Comment by ragingsquirrel3** [comment]
For this prototype, it would be useful to draft a list of facets for each category.  This will help people coordinate what data needs to be provided and ultimately integrated into search.  For example.

* genes
  * species
  * gene type (sequence ontology)
  * affected GO term
  * affected anatomy
  * phenotype
* GO
  * ??
* disease
  * ??
* ortholog groups
  * method (algorithm, e.g. panther)

**Comment by ragingsquirrel3** [comment]
Also, one of the things that came up in discussion of this was showing how orthologs/diseases annotations are derived.  Here is a little mockup of a "footnote-y" kind of thing that could be used to show how orthologs were found (this example wasn't actually via panther).  This woulc allow many different algorithms to be used and communicated to the user.  Thoughts?

![ortho_mock](https://cloud.githubusercontent.com/assets/1258388/19241867/7125731e-8ec5-11e6-954a-ae23368c0e9f.png)

**Comment by chris-grove** [comment]
@ragingsquirrel3  @monicacecilia @cmungall @doughowe 
If the point of these first mockups is to exhibit ideas for implementation of the first use case, I'd like to recommend we stay focused on the specifically what the first use case asks for:

https://docs.google.com/document/d/1M8bNyH_MIc1fHrrPJdn8AtgedvzP8pWboP1uTKejVNQ/edit

or, if the use case is insufficient, then discuss how the use case should be changed to accommodate a different vision. This way we can remain anchored on a single document that outlines what we'd like to implement. To that end, I don't think we need to concern ourselves (right now) with gene-to-disease associations, gene-to-GO associations, or pairwise gene orthology calls for this first implementation, but rather just catalog genes (and basic info including species, name, ID, symbol, synonym, type, and chromosomal position) for each AGR species, GO terms (just the terms themselves with names, IDs and synonyms), diseases (just the diseases themselves with names, IDs, and synonyms) and orthology groups (specific, named entities (PANTHER subfamilies for now)). At least, this is what we (the portal use case working group) was thinking when we drafted the first use case. The idea was that once we implement a simple search first, we could later expand to include more associations.

Should we create a new ticket for a second round of mockups? Or are we just pushing ahead with an actual implementation draft?

I still need to add specifics about facets to the use case, which I think are a good idea. We discussed this yesterday and decided that for genes we would just start with two facets: species and gene type. For Gene Ontology terms we can start with just three facets, the three main branches of the GO: biological process, molecular function, and cellular component. The idea would be then that multiple GO terms that come up as matches to the query text could be filtered down by what branch of the ontology they exist under.
